### ISSUE TITLE
update support

### ISSUE THREAD

**Issue by FabianRutten** [CarrotRub/Fit-Launcher#12]
Since a year or so Fitgirl also posts updates. It might be interesting to have the application check versions of your currently installed games with the online update.

Since the updates can mostly be incremental with much smaller additional downloads than the original game, it would be a kicker feature giving everybody the ability to get all the fitgirl games with the latest updates without redownloading the game in the app or manually installing them.

Reply in the issue if I can be of any help.

**Comment by CarrotRub** [comment]
This "updater" can technically be made but it will mostly depends on three cases : 

- Fitgirl has a standard for updates, meaning that every game has the same way of updating itself, if this is the case then it would be easy to manage and automate it.
- Every game is different or some of them need a special way to install, this would mean that we would have to allow multiple cases for each implementation of the "updater", this can be done but we would need multiple updates and would need to check many games to ensure each type of update.
- We can get each games' updater through a third party website, this would inevitably causes doubts and be scary to new users but this is a last resort in case none of the previous cases are doable.

If you want to help us, you can try to download multiple games and their updates and see how it is done, if you are okay doing so, tell me, it would be extremely helpful 👍🏻 .

Thank you, much love <3.

**Comment by FabianRutten** [comment]
I agree, the lack of a standardized update manner is blocking this feature (not a lack on fitgirls part, she aint steam).

Though it is feasible to give users an option to update their game through the launcher, meaning we'd have to parse whether or not the latest version on Fitgirl site matches the local version.

This will mean downloading the full repack and reinstalling, though it being slow and somewhat expensive (network and compute), it is better than manually going to the site doing the same thing. 

Or at least alert that a new update is available.

**Comment by CarrotRub** [comment]
This is possible, I will look into it after finishing i18n implementation and linux compatibility, I will later take a look into the different ways fitgirl update her games, if it is possible to do it, I will.

But for the full reinstalling, it will be close to impossible due to game saves that may go corrupt when automated, I haven't updated any games through fitgirl repack, thus I cannot give a sure decision, the other way as I talked about earlier, is to use third party updater found in cs.rin.ru they are usually way better and quicker for the ones I have used.
